server/src/main/java/io/crate/expression/operator/EqOperator.java

```diff
@@ -144,7 +144,7 @@ public Query toQuery(Function function, Context context) {
             return Queries.newMatchNoDocsQuery("`" + fqn + "` = null is always null");
         }
         DataType<?> dataType = ref.valueType();
-        if (dataType.id() != ObjectType.ID && ref.indexType() == IndexType.NONE) {
+        if (dataType.id() != ObjectType.ID && dataType.id() != ArrayType.ID && ref.indexType() == IndexType.NONE) {
```

Comment:
array typed refs has `IndexType = None`, but we need to continue as array internals can be indexed